### PR TITLE
Better create drive

### PR DESCRIPTION
**Pull request by Eddie-42**
As we talked about, I redesigned the create drive page a bit (but without changing the functionality). A similar update to the search rides page is coming later.

### CODE REVIEW

File: lib/drives/pages/create_drive_page.dart [hunk 1]
```diff
@@ -193,78 +194,65 @@ class _CreateDriveFormState extends State<CreateDriveForm> {
...
                 ),
-              ],
+              ),
+            ],
+          ),
+          const Divider(),
+          SizedBox(
+            width: 150,
+            child: IncrementField(
+              maxValue: 8,
```
Comment (fynsta) [inline lib/drives/pages/create_drive_page.dart:243]: let's set that max value inside the class instead

Comment (fynsta) [reply lib/drives/pages/create_drive_page.dart:243]: inside `Trip` I guess

File: lib/util/fields/increment_field.dart [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+class IncrementField extends StatefulWidget {
+  final int minValue;
+  final int maxValue;
+  final int initialValue;
+  final Widget? icon;
+  final void Function(int?)? onChanged;
+
+  const IncrementField(
+      {super.key, this.minValue = 1, this.initialValue = 1, this.maxValue = 10, this.icon, this.onChanged});
```
Comment (fynsta) [inline lib/util/fields/increment_field.dart:11]: trailing comma

File: lib/drives/pages/create_drive_page.dart [hunk 3]
```diff
@@ -193,78 +194,65 @@ class _CreateDriveFormState extends State<CreateDriveForm> {
...
+              icon: Icon(
+                Icons.chair,
+                color: Theme.of(context).colorScheme.primary,
+              ),
+              onChanged: (int? value) {
+                setState(() {
+                  _dropdownValue = value!;
```
Comment (fynsta) [inline lib/drives/pages/create_drive_page.dart:250]: dropdown value does not make much sense anymore, and is also not descriptive of what the field is for

File: lib/util/fields/increment_field.dart [hunk 4]
```diff
@@ -0,0 +1,72 @@
+import 'package:flutter/material.dart';
+
+class IncrementField extends StatefulWidget {
+  final int minValue;
+  final int maxValue;
+  final int initialValue;
+  final Widget? icon;
+  final void Function(int?)? onChanged;
```
Comment (fynsta) [inline lib/util/fields/increment_field.dart:8]: why is the function nullable?

File: lib/util/search/address_search_field.dart [hunk 5]
```diff
@@ -7,18 +7,18 @@ import 'address_suggestion.dart';
 class AddressSearchField extends StatelessWidget {
   final AddressType addressType;
   final TextEditingController controller;
-  final void Function(AddressSuggestion) onSelected;
+  final void Function(AddressSuggestion)? onSelected;
```
Comment (fynsta) [inline lib/util/search/address_search_field.dart:10]: I don't understand this change? seems unneeded

**Comment by fynsta** [comment]
Definitely like the new design!

File: lib/drives/pages/create_drive_page.dart (hunk 3 above)
Comment (Eddie-42) [reply lib/drives/pages/create_drive_page.dart]: Yeah I know, it's already fixed

File: lib/util/search/address_search_field.dart (hunk 5 above)
Comment (Eddie-42) [reply lib/util/search/address_search_field.dart:10]: Seeing this as a general class that could be used outside of our context with suggestion objects that have to be set, I see no reason why you would have to provide a function here. In most cases, a controller would suffice

Comment (fynsta) [reply lib/util/search/address_search_field.dart:10]: I'm sorry, but I don't agree - the whole thing does not make sense if we don't have `AddressSuggestion` objects - it's literally what the `SearchField` was made for

Comment (Showcas) [reply lib/util/search/address_search_field.dart:10]: Is there a case where we would need it without an object?

Comment (Eddie-42) [reply lib/util/search/address_search_field.dart:10]: Why not give users the choice? If I just want a "Choose your favorite place in the world" textfield and just wanted to store the string, I feel like I should be able to do that without needing to provide a (){} function

Comment (fynsta) [reply lib/util/search/address_search_field.dart:10]: We are not writing a library for other people, we are writing an app the we use ourselves...

**Review by Showcas — APPROVED** [review]

File: lib/util/search/start_destination_timeline.dart [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return FixedTimeline(
+      theme: CustomTimelineTheme.of(context),
+      children: <Widget>[
+        TimelineTile(
+          contents: Padding(
+            padding: const EdgeInsets.all(4.0),
```
Comment (istzen) [inline lib/util/search/start_destination_timeline.dart:29]: can we add a little bit more Padding to the left. "Das braucht Platz zum Atmen" zitat by Merri

File: lib/util/trip/trip.dart [hunk 7]
```diff
@@ -4,6 +4,8 @@ import '../model.dart';
 import '../search/position.dart';
 
 abstract class Trip extends Model {
+  static const int maxSelectableSeats = 8;
```
Comment (istzen) [inline lib/util/trip/trip.dart:7]: why did we reduce that to 8?

File: lib/drives/pages/create_drive_page.dart [hunk 8]
```diff
@@ -199,78 +199,66 @@ class _CreateDriveFormState extends State<CreateDriveForm> {
...
+              ),
+            ],
+          ),
+          const Divider(),
+          SizedBox(
+            width: 150,
+            child: IncrementField(
+              initialValue: _seats,
+              maxValue: Trip.maxSelectableSeats,
+              icon: Icon(
```
Comment (istzen) [inline lib/drives/pages/create_drive_page.dart:250]: can we make the chair icon a little bit bigger

File: lib/drives/pages/create_drive_page.dart [hunk 9]
```diff
@@ -199,78 +199,66 @@ class _CreateDriveFormState extends State<CreateDriveForm> {
...
                   ),
                 ),
-              ],
+              ),
+            ],
+          ),
+          const Divider(),
+          SizedBox(
+            width: 150,
+            child: IncrementField(
```
Comment (istzen) [inline lib/drives/pages/create_drive_page.dart:247]: can we maybe add some kind of tooltip or also a Border around that so it's clear that you are selecting the seats.

File: lib/util/trip/trip.dart (hunk 7 above)
Comment (istzen) [reply lib/util/trip/trip.dart:7]: i'll change it in the database so it's only >= 1

Comment (fynsta) [reply lib/util/trip/trip.dart:7]: Oh okay then that should be changed in this PR as well.

Although that makes the display more difficult for us... The 100 seat icons in a row would overflow completely.

Comment (Eddie-42) [reply lib/util/trip/trip.dart:7]: I think we should place some sort of limit on the number of seats. If your cat taps on the + icon 100 times, you have to click on the - icon 100 times or I have to implement a reset functionality. I think the increment field is not really designed to handle even high 2-digit numbers and at some point we should safeguard against silly inputs as well. What Fynn found isn't a car anymore, it's under the tab "trucks and vans" on the website, at which point we have to decide whether we want to accomodate 49-seater buses as well. BlaBlaCar's limit is 4, by the way!

**Review by fynsta — APPROVED** [review]

File: lib/util/trip/trip.dart (hunk 7 above)
Comment (istzen) [reply lib/util/trip/trip.dart:7]: then let's just leave 8?

File: lib/util/fields/increment_field.dart [hunk 10]
```diff
@@ -31,48 +32,52 @@ class _IncrementFieldState extends State<IncrementField> {
...
-            alignment: Alignment.centerLeft,
-            child: IconButton(
-              icon: const Icon(Icons.remove_circle_outline),
-              onPressed: _value <= widget.minValue
-                  ? null
-                  : () {
-                      _value -= 1;
-                      widget.onChanged(_value);
-                    },
+    return Card(
```
Comment (istzen) [inline lib/util/fields/increment_field.dart:35]: Maybe not a card, but a border like for the other Inputs? and for people who don't get that that should be the seats write seats somewhere (Merri did not get it right away)

Comment (fynsta) [reply lib/util/fields/increment_field.dart:35]: Hm everybody in the user study got that part right away. Should probably wait for some more feedback there?

File: lib/util/trip/trip.dart (hunk 7 above)
Comment (fynsta) [reply lib/util/trip/trip.dart:7]: Yes, can you add it in supabase again?

**Comment by fynsta** [comment]
@istzen Can we merge this from your POV? Making the seats clearer can be done after finishing up the user studies.

**Review by istzen — APPROVED** [review]

**Pull request merged**